src/main/java/core/basesyntax/factory/impl/OperationHandlerFactoryImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.factory.impl;
+
+import core.basesyntax.factory.OperationHandlerFactory;
+import core.basesyntax.model.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceStrategyImpl;
+import core.basesyntax.strategy.impl.PurchaseStrategyImpl;
+import core.basesyntax.strategy.impl.ReturnStrategyImpl;
+import core.basesyntax.strategy.impl.SupplyStrategyImpl;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationHandlerFactoryImpl implements OperationHandlerFactory {
+    private final Map<Operation, OperationHandler> operations = new HashMap<>();
+
+    public OperationHandlerFactoryImpl() {
+        operations.put(Operation.BALANCE, new BalanceStrategyImpl());
+        operations.put(Operation.RETURN, new ReturnStrategyImpl());
+        operations.put(Operation.PURCHASE, new PurchaseStrategyImpl());
+        operations.put(Operation.SUPPLY, new SupplyStrategyImpl());
+    }
```

Comment:
Just my thoughts on the original implementation of the shop, you do not have to fix the comments regarding it, unless it is necessary for writing proper tests.
I get the idea of a factory, that it produces a map for the strategy, but we have a bit of a mix in responsibilities here:
* producing the handlers
* getting the one we need based on the Operation(strategy) 

To fix that I propose to receive that map of handlers in the constructor to move the fabrication responsibility outside, and then it is not a factory anymore, but just a strategy that could be named accordingly

Reply:
I suppose It makes sense, need to process this info